hnn_core/viz.py

```diff
@@ -270,8 +270,16 @@ def plot_dipole(dpl, tmin=None, tmax=None, ax=None, layer='agg', decim=None,
 
     if isinstance(dpl, Dipole):
         dpl = [dpl]
-    elif average:
-        dpl = dpl + [average_dipoles(dpl)]
+    elif isinstance(dpl, list):
+        for single_dpl in dpl:
+            if not isinstance(single_dpl, Dipole):
+                raise ValueError('dpl should be of type Dipole, '
+                                 f'but {single_dpl } is a {type(single_dpl )}')
+        if average:
+            dpl = dpl + [average_dipoles(dpl)]
+    else:
+        raise ValueError('dpl should be of type Dipole, '
+                         f'but {dpl } is a {type(dpl )}')
```

Comment:
Let's avoid printing the whole dpl object whatever it might be.

Reply:
I've updated the error messages @rythorpe.